pkg/proxy/winkernel/proxier.go

```diff
@@ -1079,6 +1095,9 @@ func (proxier *Proxier) syncProxyRules() {
 				newHnsEndpoint.refCount = proxier.endPointsRefCount.getRefCount(newHnsEndpoint.hnsID)
 				*newHnsEndpoint.refCount++
 				svcInfo.remoteEndpoint = newHnsEndpoint
+				// store newly created endpoints in queriedEndpoints
+				queriedEndpoints[newHnsEndpoint.hnsID] = newHnsEndpoint
```

Comment:
+1, suggest to use another map to cache the (endpoint IP, endpoint ID) pair.

Reply:
I see now that the new `getAllEndpointsByNetwork` is storing by `hsnID` and `ip` but I don't see a reason why.  There is only one spot that the `queriedEndpoints` is looked up by `hnsId` and it is followed immediately by a look up for `ip`.  I think we should just store and look up by `ip` or `hsnId` unless there is a reason needed for looking up by both, which isn't immediately apparent.

Reply:
Thanks for your suggestion! Out of an abundance of caution, I did not want to change any of the logic in syncProxyRules that was used and wanted to avoid doing any optimizations outside of caching HNS information. 

But I agree this should be safe to remove. In my latest commit, I have removed the storage by endpoint ID, and am storing by endpoint IP only. Thank you!

Reply:
After more deliberation with @sbangari, I will revert back to query endpoints by ID, and thereby double the size of the map again. Unfortunately it appears there is a bug in Windows Server 2019 that could sometimes cause 2 endpoints to be populated with the same IP address, and there is fear of this potentially causing issues if we only query by IP address and remove the call to query by ID. Doubling the size of the map is acceptable for now and we will clean this up instead in the next round of optimizations coming in 2-3 months, after we had sufficient time for testing and validation.

Reply:
Out of abundance of caution, I have reverted back to the old behavior to ensure nothing breaks on Windows Server 2019.